superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -227,7 +240,8 @@ class SaveModal extends React.Component {
               onChange={this.onDashboardSelectChange}
               autoSize={false}
               value={
-                this.state.saveToDashboardId || this.state.newDashboardName
+                (this.state.saveToDashboardId ||
+                  this.state.newDashboardName) as ValueType<Record<string, any>>
```

Comment:
i'm a bit confused by this typing, isn't this `string | number`? I would've expected the Select component to be fine with that

Reply:
or is it because newDashboardName can be undefined that you need to cast this

Reply:
After further discussion, is this the same option issue as before too? Maybe we should construct an option object to pass in here instead of only passing in the string